apps/etterlatte-beregning/src/test/kotlin/avkorting/AvkortingValiderTest.kt

```diff
@@ -0,0 +1,174 @@
+                                inntektsavkorting =
+                                    listOf(
+                                        Inntektsavkorting(
+                                            avkortinggrunnlag(periode = Periode(fom = YearMonth.of(2024, 1), tom = null)),
+                                        ),
+                                    ),
+                            ),
+                        ),
+                )
+            val behandling =
+                behandling(
+                    virkningstidspunkt = virkningstidsunkt(YearMonth.of(2024, 3)),
+                    behandlingType = BehandlingType.REVURDERING,
+                )
+
+            val utenFratrekk = inntektDto(fratrekkInnAar = 0, fratrekkInnAarUtland = 0)
+            validerInntekt(utenFratrekk, avkorting, behandling)
+
+            assertThrows<ErFulltAar> {
+                val inntektMedFratrekk = inntektDto(fratrekkInnAar = 1, fratrekkInnAarUtland = 0)
+                validerInntekt(inntektMedFratrekk, avkorting, behandling)
+            }
+
+            assertThrows<ErFulltAar> {
+                val inntektMedFratrekkUtland = inntektDto(fratrekkInnAar = 0, fratrekkInnAarUtland = 1)
+                validerInntekt(inntektMedFratrekkUtland, avkorting, behandling)
+            }
+        }
+
+        @Test
+        fun `Revurdering inn i nytt år`() {
+            val avkorting =
+                Avkorting(
+                    aarsoppgjoer =
+                        listOf(
+                            aarsoppgjoer(
+                                forventaInnvilgaMaaneder = 6,
+                                aar = 2024,
+                                inntektsavkorting =
+                                    listOf(
+                                        Inntektsavkorting(
+                                            avkortinggrunnlag(periode = Periode(fom = YearMonth.of(2024, 7), tom = null)),
+                                        ),
+                                    ),
+                            ),
+                        ),
+                )
+            val behandling =
+                behandling(
+                    virkningstidspunkt = virkningstidsunkt(YearMonth.of(2025, 1)),
+                    behandlingType = BehandlingType.REVURDERING,
+                )
+
+            val utenFratrekk = inntektDto(fratrekkInnAar = 0, fratrekkInnAarUtland = 0)
+            validerInntekt(utenFratrekk, avkorting, behandling)
+
+            assertThrows<ErFulltAar> {
+                val inntektMedFratrekk = inntektDto(fratrekkInnAar = 1, fratrekkInnAarUtland = 0)
+                validerInntekt(inntektMedFratrekk, avkorting, behandling)
+            }
+
+            assertThrows<ErFulltAar> {
+                val inntektMedFratrekkUtland = inntektDto(fratrekkInnAar = 0, fratrekkInnAarUtland = 1)
+                validerInntekt(inntektMedFratrekkUtland, avkorting, behandling)
+            }
+        }
+
```

Comment:
Savner en case hvor valideringen faktisk er gyldig også for å ha mulighet til å sjekke at vi ikke senere brekker "happy-path" med andre endringer.

Reply:
Testet det allede i noen av testene som var der men lagde en egen eksplisitt test for det også.